### PR TITLE
Re-enable .NET Core Tests (Resolves #1617)

### LINKED ISSUE #1617: Re-enable .NET Core Tests
LINQ Expressions and Dynamic are broken in the latest .NET Core builds. We've disabled the tests in 033ee73538d63d43fdcd77b505c7e19a014d1ad1, 41079ee6f4488423822b3f62d99142594862b33d & 6c590e7d4f71db016cd854e9169360105e04332f. We should remember to re-enable them when it's fixed. :stuck_out_tongue_winking_eye:

### PR DESCRIPTION
**Pull request by bricelam**
This reverts commits 033ee73538d63d43fdcd77b505c7e19a014d1ad1, 41079ee6f4488423822b3f62d99142594862b33d & 6c590e7d4f71db016cd854e9169360105e04332f.

Note, there are two test failures that we need to investigate. For now, the tests are skipped (outputs yellow during build).

### CODE REVIEW

**Comment by anpete** [comment]
:shipit:

File: test/EntityFramework.CrossStore.FunctionalTests/EndToEndTest.cs [hunk 1]
```diff
@@ -114,6 +114,13 @@ public SharedInMemoryEndToEndTest(SharedCrossStoreFixture fixture)
             : base(fixture)
         {
         }
+
+#if DNXCORE50
```
Comment (divega) [inline test/EntityFramework.CrossStore.FunctionalTests/EndToEndTest.cs]: Are we keeping the bug open for the remaining skipped tests or filing a new one? @maumar would be good to investigate these.

Comment (maumar) [inline test/EntityFramework.CrossStore.FunctionalTests/EndToEndTest.cs]: I don't mind either way. I can file a new bug or you can assign the current one to me.

Comment (bricelam) [inline test/EntityFramework.CrossStore.FunctionalTests/EndToEndTest.cs]: I'll file a new bug. The existing issue can remain just about re-enabling all (but 2) of them.

Comment (bricelam) [inline test/EntityFramework.CrossStore.FunctionalTests/EndToEndTest.cs]: Filed #1905

**Pull request merged**